websites/C/Codeberg/presence.ts

```diff
@@ -0,0 +1,230 @@
+			profileName = document
+				.querySelector(".profile-avatar-name")
+				.querySelector(".username")
+				.textContent.split(" ")[0];
+
+		let name: string;
+		if (!profileDisplayName) name = profileName;
+		else name = `${profileDisplayName} (${profileName})`;
+
+		const tabParam = new URL(href).searchParams.get("tab");
+
+		if (tabParam) {
+			switch (tabParam) {
+				case "stars": {
+					presenceData.details = `Viewing ${profileName}'s starred repositories`;
+
+					break;
+				}
+				case "activity": {
+					presenceData.details = `Viewing ${profileName}'s public activity`;
+
+					break;
+				}
+				case "repositories": {
+					presenceData.details = `Viewing ${profileName}'s repositories`;
+
+					break;
+				}
+				// No default
+			}
+		} else if (href.includes("/-/projects"))
+			presenceData.details = `Viewing ${profileName}'s projects`;
+		else if (href.includes("/-/packages"))
+			presenceData.details = `Viewing ${profileName}'s packages`;
+		else presenceData.details = `Viewing profile: ${name}`;
+
+		presenceData.buttons = [
+			{
+				label: "View profile",
+				url: href,
+			},
+		];
+		presenceData.largeImageKey = document
+			.querySelector("#profile-avatar")
+			.querySelector("img").src;
+		presenceData.smallImageKey = Assets.Logo;
+	}
+
+	// Handle repos
+	if (
+		document.querySelector(".repository") &&
+		!document.querySelector(".repository.new")
+	) {
+		const repoAvatar = document
+				.querySelector(".flex-item-leading")
+				.querySelector("img"),
+			repoTitle = document
+				.querySelector(".flex-item-main")
+				.querySelector(".flex-item-title")
+				.textContent.trim();
+
+		if (repoAvatar) {
+			presenceData.largeImageKey = repoAvatar.src;
+			presenceData.smallImageKey = Assets.Logo;
+		} else presenceData.largeImageKey = Assets.Logo;
+
+		presenceData.buttons = [
+			{
+				label: "View repository",
+				url: href,
+			},
+		];
+
+		const repoPages: Record<string, PresenceData> = {
+			"": {
+				details: `Viewing repository: ${repoTitle}`,
+			},
+			issues: {
+				details: `Viewing ${repoTitle}'s issues`,
+			},
+			pulls: {
+				details: `Viewing ${repoTitle}'s pull requests`,
+			},
+			activity: {
+				details: `Viewing ${repoTitle}'s public activity`,
+			},
+			settings: {
+				details: `Editing ${repoTitle}'s settings`,
+			},
+			releases: {
+				details: `Viewing ${repoTitle}'s releases`,
+			},
+			projects: {
+				details: `Viewing ${repoTitle}'s projects`,
+			},
+			actions: {
+				details: `Viewing ${repoTitle}'s actions`,
+			},
+			packages: {
+				details: `Viewing ${repoTitle}'s packages`,
+			},
+			branches: {
+				details: `Viewing ${repoTitle}'s branches`,
+			},
+			tags: {
+				details: `Viewing ${repoTitle}'s tags`,
+			},
+			forks: {
+				details: `Viewing ${repoTitle}'s forks`,
+			},
+			watchers: {
+				details: `Viewing ${repoTitle}'s watchers`,
+			},
+			stars: {
+				details: `Viewing ${repoTitle}'s stargazers`,
+			},
+		};
```

Comment:
Why not just use queryselector like this:
`document.querySelector('.active.item')`

Reply:
Well for most of these `*Pages` constants I'm not able to specify able to use that in the details (for instance, under `stars` I say `stargazers` and not `stars` so I can't really use querySelector like that)

Though I think the code you've written can be used as a replacement for the `if (pathname.includes(...))` conditions